IPython/core/interactiveshell.py

```diff
@@ -2826,6 +3005,8 @@ def run_code(self, code_obj, result=None):
           A compiled code object, to be executed
         result : ExecutionResult, optional
           An object to store exceptions that occur during execution.
+        async :  Bool (Experimental)
```

Comment:
Good catch – the plan is indeed to make `async` into a real keyword in 3.7, so it shouldn't be used as a name.